ui/src/components/popovers/PasteModal.tsx

```diff
@@ -56,49 +55,62 @@ export default function PasteModal() {
     setPasteMode(false);
     setDisableVideoFocusTrap(false);
     if (rpcDataChannel?.readyState !== "open" || !TextAreaRef.current) return;
-    if (!keyboardLayout) return;
-    if (!chars[keyboardLayout]) return;
-
+    const keyboard: KeyboardLayout = selectedKeyboard(keyboardLayout);
     const text = TextAreaRef.current.value;
 
     try {
       for (const char of text) {
-        const { key, shift, altRight, deadKey, accentKey } = chars[keyboardLayout][char]
-        if (!key) continue;
-
-	const keyz = [ keys[key] ];
-	const modz = [ modifierCode(shift, altRight) ];
-
-	if (deadKey) {
-            keyz.push(keys["Space"]);
-            modz.push(noModifier);
-	}
-	if (accentKey) {
-            keyz.unshift(keys[accentKey.key])
-            modz.unshift(modifierCode(accentKey.shift, accentKey.altRight))
-	}
-
-	for (const [index, kei] of keyz.entries()) {
-          await new Promise<void>((resolve, reject) => {
-            send(
-              "keyboardReport",
-              hidKeyboardPayload([kei], modz[index]),
-              params => {
-                if ("error" in params) return reject(params.error);
-                send("keyboardReport", hidKeyboardPayload([], 0), params => {
-                  if ("error" in params) return reject(params.error);
-                  resolve();
-                });
-              },
-            );
-          });
-	}
+        const keyprops = keyboard.chars[char];
+        if (!keyprops) {
+          console.log("No translation for: ", char);
+          continue;
+        }
+
+        const { key, shift, altRight, deadKey, accentKey } = keyprops;
+        if (!key) {
+          console.log("No key for: ", char);
```

Comment:
This also could be eliminated, but very useful in finding missing keystroke mappings.